### PR TITLE
 Add support for handleCardAction

### PR DESCRIPTION
**Pull request by bombek92**
### Summary & motivation

Added support for https://stripe.com/docs/stripe-js/reference#stripe-handle-card-action

### API review

**Added new method, not breaking existing code.**

Copy [this template] **or** link to an API review issue.

[this template]:
  https://github.com/stripe/react-stripe-elements/tree/master/.github/API_REVIEW.md

### Testing & documentation

**Simple unit tests calling this method.**

**YES**

### CODE REVIEW

**Comment by tremby** [comment]
Is there any workaround to use this feature before this patch is merged?

**Comment by bombek92** [comment]
> 
> 
> Is there any workaround to use this feature before this patch is merged?

@tremby Yes there is, I will edit this comment tomorrow and post it.

This PR won't be merged at this state as it is not passing the CI.

**Comment by bombek92** [comment]
> Is there any workaround to use this feature before this patch is merged?

```
  private handleCardAction = async (clientSecret: string, paymentIntentId: string) => {
    const result = await **(this.props.stripe as any)**.handleCardAction(clientSecret);
    (...)
  }
```

**Comment by tamas-norbert-miklos** [comment]
Approximately when can we expect this feature to be available? The SCA rule will come into effect in Europe on September 14. So, like in 2 weeks we won't be able to use Stripe on our project for payments...

**Comment by poteirard** [comment]
I'm happy to be the one giving the good news to you. The new regulation has been postponed 18 months [source](https://www.ft.com/content/64a738ba-bdd0-11e9-89e2-41e555e96722)

**Comment by bombek92** [comment]
> I'm happy to be the one giving the good news to you. The new regulation has been postponed 18 months [source](https://www.ft.com/content/64a738ba-bdd0-11e9-89e2-41e555e96722)

FYI This is UK only.

**Comment by tamas-norbert-miklos** [comment]
> > I'm happy to be the one giving the good news to you. The new regulation has been postponed 18 months [source](https://www.ft.com/content/64a738ba-bdd0-11e9-89e2-41e555e96722)
> 
> FYI This is UK only.

Our projects it's for Italy. So i guess the original deadline still applies.

**Comment by giupea** [comment]
> > > I'm happy to be the one giving the good news to you. The new regulation has been postponed 18 months [source](https://www.ft.com/content/64a738ba-bdd0-11e9-89e2-41e555e96722)
> > 
> > 
> > FYI This is UK only.
> 
> Our projects it's for Italy. So i guess the original deadline still applies.

Deadline in Italy seems to have been postponed (no new date set): https://www.bancaditalia.it/media/comunicati/documenti/2019-02/CS_01082019_a_f.pdf (italian)

**Comment by yamsellem** [comment]
@bombek92 it seems that the travis build is failing. May you please take a look why? With a proper build, this feature will get better change to be merged quickly (I hope it'll, it should help a lot). Thanks.

**Comment by yamsellem** [comment]
@bombek92 meanwhile your code is merged, I don't understand your answer on a temporary workaround. Where are you suggesting to add the code you suggest to @tremby?

Because `react-stripe-elements` is a typescript package, it does not seems possible to use your fork as a npm dependency (adding `"react-stripe-elements": "bombek92/react-stripe-elements"` to a package.json dependencies does not work).

Thanks again.

**Comment by bombek92** [comment]
> @bombek92 meanwhile your code is merged, I don't understand your answer on a temporary workaround. Where are you suggesting to add the code you suggest to @tremby?
> 
> Because `react-stripe-elements` is a typescript package, it does not seems possible to use your fork as a npm dependency (adding `"react-stripe-elements": "bombek92/react-stripe-elements"` to a package.json dependencies does not work).
> 
> Thanks again.

just use:
`const result = await (this.props.stripe as any).handleCardAction(clientSecret);`

**Comment by bombek92** [comment]
> @bombek92 it seems that the travis build is failing. May you please take a look why? With a proper build, this feature will get better change to be merged quickly (I hope it'll, it should help a lot). Thanks.

That's why this is draft pr, but I'll try to fix it this weekend.

**Comment by yamsellem** [comment]
@bombek92 Thanks for your quick answers.

In my client side code, I use react, but in javascript file (no typescript) and with promise (no await), is your workaround still possible with this, and what would be the code?

Thanks again.

**Comment by yamsellem** [comment]
@bombek92 have you found some time to take a look? You would be a life saver if you do. Thanks (14 of September makes me worry to death).
